evil-commands.el

```diff
@@ -636,6 +637,29 @@ and jump to the corresponding one."
           (when ov
             (goto-char (overlay-start ov))))))))
 
+(defun evil-ispell-mark-word-as-good (word)
+  "Add WORD under the cursor as a good word to the Ispell dictionary."
+  (interactive (list (thing-at-point 'word)))
+  (ispell-send-string (concat "*" word "\n"))
+  ;; dictionary modified!
+  (setq ispell-pdict-modified-p '(t))
+  (ispell-pdict-save ispell-silently-savep)
+  (when (bound-and-true-p flyspell-mode)
+    (let ((start (car (bounds-of-thing-at-point 'word))))
+      (and start
+           (flyspell-unhighlight-at start))))
+  (message "Word '%s' added to dictionary" (upcase word)))
+
+(defun evil-ispell-mark-word-as-locally-good (word)
+  "Add WORD under the cursor to the local word list."
+  (interactive (list (thing-at-point 'word)))
+  (ispell-add-per-file-word-list word)
+  (when (bound-and-true-p flyspell-mode)
+    (let ((start (car (bounds-of-thing-at-point 'word))))
+      (and start
```

Comment:
Use `when`, not `and` when dealing with side effects.

Reply:
Updated.